### PR TITLE
[BEAM-498] Update DoFn javadocs to remove references to OldDoFn and Dataflow

### PR DESCRIPTION
**Pull request by swegner**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Various components had references to `OldDoFn` where it would be more appropriate to describe the "new" `DoFn`.

Along the way, updated a few Dataflow references to Beam.

### CODE REVIEW

**Comment by swegner** [comment]
R: @tgroh @kennknowles

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java [hunk 1]
```diff
@@ -416,7 +416,7 @@ void invokeFinishBundle(
         DoFn<InputT, OutputT>.Context c,
         ExtraContextFactory<InputT, OutputT> extra);
 
-    /** Invoke {@link OldDoFn#teardown()} on the bound {@code DoFn}. */
+    /** Invoke {@link OldDoFn#teardown()} on the bound {@code OldDoFn}. */
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java]: This seems more for invoking the annotated methods on the [Nu]DoFn

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java [hunk 2]
```diff
@@ -72,45 +72,45 @@
...
   /**
    * Returns a {@code DoFnTester} supporting unit-testing of the given
-   * {@link OldDoFn}.
+   * {@link DoFn}.
    */
   @SuppressWarnings("unchecked")
-  public static <InputT, OutputT> DoFnTester<InputT, OutputT> of(OldDoFn<InputT, OutputT> fn) {
-    return new DoFnTester<InputT, OutputT>(fn);
+  public static <InputT, OutputT> DoFnTester<InputT, OutputT>
+  of(DoFn<InputT, OutputT> fn) {
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: Strange new linebreak

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java [hunk 3]
```diff
@@ -182,7 +182,7 @@ public CloningBehavior getCloningBehavior() {
   }
 
   /**
-   * Calls {@link OldDoFn#startBundle} on the {@code OldDoFn} under test.
+   * Calls the {@link DoFn.StartBundle} on the {@link DoFn} under test.
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: either "the StartBundle method" or "StartBundle"

I prefer the former, but up2u

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java [hunk 4]
```diff
@@ -182,7 +182,7 @@ public CloningBehavior getCloningBehavior() {
...
    *
    * <p>If needed, first creates a fresh instance of the OldDoFn under test.
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java]: Missed?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/SystemDoFnInternal.java [hunk 5]
```diff
@@ -31,7 +31,7 @@
  * <p>Currently, the only effect of this is to mark any aggregators reported by an annotated
  * {@code OldDoFn} as a system counter (as opposed to a user counter).
```
Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/SystemDoFnInternal.java]: Missed?

**Comment by swegner** [comment]
Rebased onto latest changes from master and including fixes for previous feedback
R: @kennknowles for committer approval and merge.

**Comment by kennknowles** [comment]
Error in Jenkins seems unrelated, but possibly trouble. Like autovalue is no longer configured properly?

**Comment by swegner** [comment]
Turns out failures were due to a bad rebase. Latest version builds locally.

**Comment by kennknowles** [comment]
LGTM. Merging.

**Pull request merged**